src/python-bindings/common2/py_util.cpp

```diff
@@ -1 +1,265 @@
+PyObject_Handle *
+get_handle_from(PyObject * py) {
+	auto * py_handle = PyObject_GetAttrString( py, "_handle" );
+	Py_DecRef(py_handle);
+	return (PyObject_Handle *)py_handle;
+}
 
+
+PyObject *
+py_new_htcondor2_classad(void * classAd) {
+	static PyObject * py_htcondor2_module = NULL;
+	if( py_htcondor2_module == NULL ) {
+		 py_htcondor2_module = PyImport_ImportModule( "htcondor2" );
+	}
+
+	static PyObject * py_htcondor2_classad_module = NULL;
+	if( py_htcondor2_classad_module == NULL ) {
+		py_htcondor2_classad_module = PyObject_GetAttrString( py_htcondor2_module, "classad" );
+	}
+
+	static PyObject * py_ClassAd_class = NULL;
+	if( py_ClassAd_class == NULL ) {
+		py_ClassAd_class = PyObject_GetAttrString( py_htcondor2_classad_module, "ClassAd" );
+	}
+
+	PyObject * pyClassAd = PyObject_CallObject(py_ClassAd_class, NULL);
+	auto * handle = get_handle_from(pyClassAd);
+
+	// PyObject_CallObject() should have called __init__() for us, which
+	// should call _classad_init(), which should have made sure that the
+	// Python ClassAd object is valid by creating a new ClassAd for it.
+	if( classAd != NULL ) {
+		// Arguably, for tracing purposes, we should call handle->f() here.
+		if( handle->t != NULL ) { delete (ClassAd *)handle->t; }
```

Comment:
Add a comment here clarifying that the ClassAd constructor is expected to create an empty C++ ClassAd and stuff it into handle->t.